jupyterhub/scopes.py

```diff
@@ -143,6 +143,36 @@
     'access:services': {
         'description': 'Access services via API or browser.',
     },
+    'users:shares': {
+        'description': "Read and revoke a user's access to shared servers.",
+        'subscopes': [
+            'read:users:shares',
+        ],
+    },
```

Comment:
Isn't being granted `users:shares` being granted the ability to read/create/revoke any user's shares, and not just a user and not just read/revoke?

I was thinking maybe a filter is what limits this permission to a speific user, but this permission is described without a filter applied as a whole now.

Reply:
Hm, I'm never quite satisfied with scope descriptions and combining with filters. All scopes are described assuming no filter, and then filters limit e.g. _which_ user can be acted upon.

Having this scope means you can take the _action_ "read and revoke a user's access to shared servers", which is the prose representation of `GET|DELETE /api/users/:name/shared[/:server]`, while the filter specifies _which_ users you can take that action upon. If you have a clearer summary that remains concise, I think there's always room for improvement here.